src/3d/qgs3dmapscene.cpp

```diff
@@ -592,6 +592,12 @@ void Qgs3DMapScene::addLayerEntity( QgsMapLayer *layer )
     QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer );
     connect( vlayer, &QgsVectorLayer::selectionChanged, this, &Qgs3DMapScene::onLayerRenderer3DChanged );
   }
+
+  if ( layer->type() == QgsMapLayerType::MeshLayer )
+  {
+    connect( layer, &QgsMapLayer::rendererChanged, this, &Qgs3DMapScene::onLayerRenderer3DChanged );
```

Comment:
Before this PR, only vector layer need this. Now there is too mesh layer. I don't now the reason why it is needed for vector layer, but it is needed for mesh layer when the dataset or the color ramp shader of the contour change. 
It is a quite heavy operation because, the entity need to be rebuilt and with this connection, all the times the 2D renderer changes.
I am afraid that for other layer type, it could be heavy task for nothing.

Reply:
ok